.github/workflows/codemod_publish.yml

```diff
@@ -1,107 +1,82 @@
+          # Extract version (everything before @)
+          VERSION="${TAG%@*}"
+          VERSION="${VERSION#v}"  # Remove 'v' prefix
 
-      - name: Checkout Code
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
-        with:
-          fetch-depth: 0
+          # Extract codemod name (everything after @)
+          CODEMOD_NAME="${TAG#*@}"
 
-      - name: Set matrix for codemods
-        id: set-matrix
-        run: |
-          FILES_JSON=$(echo '${{ needs.paths-filter.outputs.codemods_files }}' | jq -c '{include: map({file: .})}')
-          echo "codemod_files=$FILES_JSON" >> "$GITHUB_OUTPUT"
-          echo "Matrix JSON: $FILES_JSON"
+          echo "VERSION=$VERSION" >> $GITHUB_OUTPUT
+          echo "CODEMOD_NAME=$CODEMOD_NAME" >> $GITHUB_OUTPUT
+          echo "CODEMOD_PATH=recipes/$CODEMOD_NAME" >> $GITHUB_OUTPUT
 
-  publish:
-    name: Publish ${{ matrix.file }}
-    runs-on: ubuntu-latest
-    needs: prepare-matrix
-    strategy:
-      fail-fast: false
-      matrix: ${{fromJson(needs.prepare-matrix.outputs.codemod_files)}}
-    env:
-      CODEMOD_API_KEY: ${{ secrets.CODEMOD_API_KEY }}
-    steps:
-      - name: Harden the runner (Audit all outbound calls)
-        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49
-        with:
-          egress-policy: audit
+          echo "Publishing: $CODEMOD_NAME (version: $VERSION) from recipes/$CODEMOD_NAME"
 
-      - name: Checkout Code
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
-        with:
-          fetch-depth: 0
+      - name: Validate codemod directory exists
+        run: |
+          if [ ! -d "${{ steps.extract.outputs.CODEMOD_PATH }}" ]; then
+            echo "Error: Directory ${{ steps.extract.outputs.CODEMOD_PATH }} does not exist"
+            exit 1
+          fi
 
-      - uses: actions/setup-node@cdca7365b2dadb8aad0a33bc7601856ffabcc48e
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
           node-version-file: ".nvmrc"
+          cache: 'npm'
 
       - name: Install dependencies
-        run: |
-          sudo apt-get install libsecret-1-dev
-          npm install -g codemod
+        run: npm ci
 
-      - name: Test codemod
-        run: |
-          DIR=$(dirname "${{ matrix.file }}")
-          echo "Testing codemod in: $DIR"
-          cd "$DIR"
-          npm install
-          npm run --if-present test
+      - name: Install Codemod CLI
+        run: npm install -g codemod@next
+
+      - name: Login to Codemod registry
+        env:
+          CODEMOD_TOKEN: ${{ secrets.CODEMOD_TOKEN }}
+        run: codemod login --token "$CODEMOD_TOKEN"
+
+      - name: Run test
+        working-directory: ${{ steps.extract.outputs.CODEMOD_PATH }}
+        run: node --run test
+
+      - name: Publish codemod (dry run)
+        working-directory: ${{ steps.extract.outputs.CODEMOD_PATH }}
+        run: codemod publish --dry-run
```

Comment:
Why a dry-run right before the full?

Reply:
because the cli isn't 100% stable i want to run one time to unsere we will not publish an ivalid tarball because cli don't cut the process before sending it 

(BTW CLI din't have any test)